packages/core/src/components/menu/menuItem.tsx

```diff
@@ -118,8 +113,8 @@ export class MenuItem extends AbstractComponent<IMenuItemProps, IMenuItemState>
                 tabIndex={disabled ? undefined : 0}
                 target={this.props.target}
             >
-                {labelElement}
-                {this.props.text}
+                <span className={Classes.MENU_ITEM_TEXT}>{this.props.text}</span>
+                {label != null && <span className={Classes.MENU_ITEM_LABEL}>{label}</span>}
```

Comment:
in this codebase, we prefer to use ternary `condition ? <el /> : undefined` instead of boolean coercion.